src/App.tsx

```diff
@@ -1,61 +1,149 @@
+import { useEffect, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import { ToDo } from './types/ToDo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const todosInfo: ToDo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosInfo);
+  const [user, setUser] = useState('');
+  const [title, setTitle] = useState('');
+  const [hasUser, setHasUser] = useState(false);
+  const [hasTitle, setHasTitle] = useState(false);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+
+  useEffect(() => {
+    if (title) {
+      setHasTitle(true);
+    }
+
+    if (user) {
+      setHasUser(true);
+    }
+  }, [title, user]);
+
+  function handleChange(e: React.BaseSyntheticEvent) {
+    const { name } = e.target;
+
+    if (name === 'title') {
+      setTitle(e.target.value);
+    }
+
+    if (name === 'user') {
+      setUser(e.target.value);
+    }
+  }
+
+  function handleSubmit(e: React.FormEvent) {
+    e.preventDefault();
+
+    setFormSubmitted(true);
+
+    if (!title) {
```

Comment:
```suggestion
    if (!title.trim()) {
```

Reply:
previous comment was not fixed yet
```suggestion
    if (!title.trim()) {
```